File: 319/bulb-switcher.js
/**
 * @param {number} n
 * @return {number}
 */
var bulbSwitch = function (n) {
    return intSqrt(n);
};

var intSqrt = function (x) {
    let root = x;
    while (root * root > x) {
        root = Math.floor((root + x / root) / 2);
    }
    return root;
};

let n = 3;
console.log(bulbSwitch(n));

File: 319/bulb-switcher.go
package main

import "fmt"

func bulbSwitch(n int) int {
	return intSqrt(n)
}

func intSqrt(x int) int {
	root := x

	for root*root > x {
		root = (root + x/root) / 2
	}

	return root
}

func main() {
	n := 3
	fmt.Println(bulbSwitch(n))
}
